### PR TITLE
docs: add Node.js API update in migration guide

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [x] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
No

### Motivation / Use-Case
update migration guide

### Breaking Changes
none

### Additional Info
no

### CODE REVIEW

**Comment by snitin315** [comment]
I think we should also add a deprecation warning here - 

https://github.com/webpack/webpack-dev-server/blob/fe8dda7b4f5f2c5ff52864288a64058ed7e3187e/lib/Server.js#L21-L24

**Comment by alexander-akait** [comment]
Yes, make sense

**Comment by snitin315** [comment]
I will add 👍

File: migration-v4.md [hunk 1]
```diff
@@ -500,3 +500,13 @@ There are a lot of other bug fixes.
   ```
 
   IE8 is not supported, sorry
+
+- Change in **Node.js API**:
+
+  - If you're using dev-server through the Node.js API, the options in devServer will be ignored. Pass the options as a first parameter instead:
+
+  ```js
+  new WebpackDevServer(compiler, {...})
```
Comment (alexander-akait) [inline migration-v4.md:509]: Let's add v3/v4 as we done before

**Review by alexander-akait — APPROVED** [review]

**Pull request merged**